assets/js/googlesitekit/datastore/user/user-input-settings.js

```diff
@@ -172,12 +201,63 @@ const baseSelectors = {
 	 * @since 1.19.0
 	 *
 	 * @param {Object} state Data store's state.
-	 * @return {(Object|undefined)} User input settings.
+	 * @return {(Array.<string>|undefined)} User input setting values.
 	 */
 	getUserInputSetting: createRegistrySelector( ( select ) => ( state, settingID ) => {
 		const settings = select( STORE_NAME ).getUserInputSettings() || {};
-		return settings[ settingID ];
+		const values = settings[ settingID ]?.values;
+		return Array.isArray( values ) ? values : [];
+	} ),
+
+	/**
+	 * Gets a scope of the input setting.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(string|undefined)} User input setting scope.
+	 */
+	getUserInputSettingScope: createRegistrySelector( ( select ) => ( state, settingID ) => {
+		const settings = select( STORE_NAME ).getUserInputSettings() || {};
+		return settings[ settingID ]?.scope;
 	} ),
+
+	/**
+	 * Gets an author of the input setting.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(Object|undefined)} User input setting author.
+	 */
+	getUserInputSettingAuthor: createRegistrySelector( ( select ) => ( state, settingID ) => {
+		const settings = select( STORE_NAME ).getUserInputSettings() || {};
+		return settings[ settingID ]?.author;
+	} ),
+
+	/**
+	 * Determines whether or not user input settings have been started saving.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {boolean} TRUE if saving has started, otherwise FALSE.
+	 */
+	hasStartedSavingInputSettings( state ) {
+		return !! state.hasStartedSavingInputSettings;
+	},
+
+	/**
+	 * Determines whether or not user input settings have been saved.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {boolean} TRUE if saving has finished, otherwise FALSE.
+	 */
+	hasFinishedSavingInputSettings( state ) {
+		return !! state.hasFinishedSavingInputSettings;
+	},
```

Comment:
See above, I think we don't need these selectors.

Reply:
It is still needed because it helps to identify when we need to redirect a user to the dashboard.

Reply:
I think this should be handled based on the submit user action, see my above comment. So these two selectors can be removed.

Reply:
Yep, you are right. I got it now. removed these selectors.